modules/libcom/RTEMS/posix/rtems_init.c

```diff
@@ -938,6 +929,27 @@ POSIX_Init ( void *argument __attribute__((unused)))
     struct timespec  	now;
     char timeBuff[100];
 
+    /*
+     * If RTEMS is used with the POSIX API, the init task
+     * 'POSIX_Init()' is unfortunately given the priority '2'. This
+     * corresponds to the second lowest POSIX prio (RTEMS pthread
+     * prio 254). This task shoud have IOCsh prio.
+     */
+    pthread_attr_t attr;
+    struct sched_param  param;
+    int policy;
+    sc  = pthread_attr_init(&attr);
+    assert(sc == RTEMS_SUCCESSFUL);
+    sc = pthread_attr_getschedpolicy(&attr, &policy);
+    assert(sc == RTEMS_SUCCESSFUL);
+
+    param.sched_priority = (sched_get_priority_max(policy)
+                            - sched_get_priority_min(policy))
+                         * epicsThreadPriorityIocsh / 100;
```

Comment:
@hjunkes I'm sorry to keep poking at this, but it seems like the basics of the linear equation.  I see the slope.  Where is the offset?

Consider the edge cases.

if `epicsThreadPriorityIocsh=0`, then the result needs to be `sched_get_priority_min(policy)`.

If `epicsThreadPriorityIocsh=100`, then the result needs to be `sched_get_priority_max(policy)`.

I think you need to change this expression to:

```c
    param.sched_priority = (sched_get_priority_max(policy) - sched_get_priority_min(policy))
                         * epicsThreadPriorityIocsh / 100
                         + sched_get_priority_min(policy);
```

Reply:
Yes, you are absolutely right, I have been sloppy here.